rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,258 @@
+
+### Observing pipelines
+
+Users should be able to observe and monitor individual pipelines.
+This means relevant metrics coming from the `internal_metrics` source must contain a `pipeline_id` tag referring to the pipeline `id`.
+
+In Vector, the `Task` structure is what emits the events for `internal_metrics`.
+After [build the different pieces of the topology](https://github.com/timberio/vector/blob/v0.15.0/src/topology/builder.rs#L106), we've to update the [`Task::new`](https://github.com/timberio/vector/blob/v0.15.0/src/topology/builder.rs#L163) in order to accept an `Option<PipelineId>` so that when it emits the metrics events it can provide the information about the pipeline.
+
+This approach would extend the [RFC 2064](https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md#collecting-uniform-context-data) by _just_ adding `pipeline_id` to the context.
+
+When [spawning a transform](https://github.com/timberio/vector/blob/v0.15.0/src/topology/mod.rs#L574), adding the optional pipeline information to the span will populate the metrics.
+
+```rust
+let span = error_span!(
+    "transform",
+    component_kind = "transform",
+    component_name = %task.name(),
+    component_type = %task.typetag(),
+    pipeline_id = %task.pipeline_id(),
+);
+```
+
+## Rationale
+
+- Why is this change worth it?
+
+This split improves the readability of the configuration files and allows the users to collaborate, which makes using Vector more user friendly.
+
+- What is the impact of not doing this?
+
+This would force the users to keep having complex configuration files and/or to duplicate components configuration between their configuration files.
+
+- How does this position us for success in the future?
+
+
+## Prior Art
+
+_TODO_
+
+## Drawbacks
+
+- Why should we not do this?
+- What kind on ongoing burden does this place on the team?
+
+## Alternatives
+
+- Do nothing: we can already use several configuration files, people could split their existing configuration.
+
+This would imply some duplication if a transform is used in multiple configuration files.
+Anybody could add a sink or source even if they don't have permission.
+
+- Do nothing: write a tool that would write a big configuration file where each pipeline would start with a dummy filter that we could monitor in the `internal_metrics`.
+
+
+
+- Evolve vector to use a tag/filter model like our competitors, have a 'pipeline' be a 'tag'.
+
+Not able to add internal metrics to specific transforms and monitor them.
+Doesn't block to create other sources/sinks.
+
+- Run a single vector per-'pipeline' and support metric tagging to distinguish at the telemetry level.
+
+Adds lot of complexity and would add some constraints regarding resources that can only been used once.
+Doesn't block to create other sources/sinks.
+
+## Outstanding Questions
```

Comment:
@lukesteensen didn't you have other questions to add?

- What is the default pipeline?
- What about pipelines that don't have transforms (sources connecting directly to sinks)

Anything else?